pg-pong.py

```diff
@@ -0,0 +1,147 @@
+""" Trains an agent with (stochastic) Policy Gradients on Pong. Uses OpenAI Gym.
+    From http://karpathy.github.io/2016/05/31/rl/. Additional comments added, variables renamed. """
+import numpy as np
+import cPickle as pickle
+import gym
+
+# hyperparameters
+H = 200 # number of hidden layer neurons
+batch_size = 10 # every how many episodes to do a param update?
+learning_rate = 1e-4
+gamma = 0.99 # discount factor for reward
+decay_rate = 0.99 # decay factor for RMSProp leaky sum of grad^2
+resume = True # resume from previous checkpoint?
+render = True
+
+# model initialization
+D = 80 * 80 # input dimensionality: 80x80 grid
+if resume:
+  model = pickle.load(open('save.p', 'rb'))
+else:
+  model = {}
+  model['W1'] = np.random.randn(H,D) / np.sqrt(D) # "Xavier" initialization
+  model['W2'] = np.random.randn(H) / np.sqrt(H)
+  
+grad_buffer = { k : np.zeros_like(v) for k,v in model.iteritems() } # update buffers that add up gradients over a batch
```

Comment:
Python 3:

    grad_buffer = { k : np.zeros_like(v) for k,v in model.items() }